### PR TITLE
Week6

### PR DESCRIPTION
**Pull request by kulloveth**

### CODE REVIEW

File: Week4/app/src/main/java/com/kulloveth/moviesapp/db/repository/RoomRepository.kt [hunk 1]
```diff
@@ -20,20 +21,23 @@ class RoomRepository : MovieRepository {
         dao.deleteMovie(movieEntity)
     }
 
-    override fun getAllMovies(): LiveData<List<Movie>> {
+    override fun getAllMovies(): Flow<List<Movie>> {
```
Comment (jenndotcodes) [inline Week4/app/src/main/java/com/kulloveth/moviesapp/db/repository/RoomRepository.kt:24]: Nice work using Flow.

File: Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt [hunk 2]
```diff
@@ -46,9 +47,8 @@ abstract class MovieDatabse : RoomDatabase() {
         ).addCallback(object : RoomDatabase.Callback() {
             override fun onCreate(db: SupportSQLiteDatabase) {
                 super.onCreate(db)
-                GlobalScope.launch(context = Dispatchers.IO) {
```
Comment (jenndotcodes) [inline Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt:49]: The assignment said not to use GlobalScope, is there another way to do this?

File: Week4/app/src/main/java/com/kulloveth/moviesapp/ui/MoviesDataManager.kt [hunk 3]
```diff
@@ -27,47 +26,66 @@ class MoviesDataManager(application: Application) : AndroidViewModel(application
...
-    * sorts  the movies and arrange
-    * according to its genres
+    * fetch movies from room using flow
+    * and convert to livedata
     * */
 
-
     fun getMovieComposites(): LiveData<List<CompositeItem>> {
-        var moviesList: MutableList<Movie>
-        viewModelScope.launch(Dispatchers.IO) {
```
Comment (jenndotcodes) [inline Week4/app/src/main/java/com/kulloveth/moviesapp/ui/MoviesDataManager.kt:47]: Nice job using the viewModelScope.

File: Week4/app/src/main/java/com/kulloveth/moviesapp/ui/movies/MoviesFragment.kt [hunk 4]
```diff
@@ -57,52 +65,36 @@ class MoviesFragment : Fragment(), MovieAdapter.MovieItemCLickedListener {
...
         (requireActivity() as AppCompatActivity?)?.setSupportActionBar(binding?.contentLayout?.toolbar)
+
         moviesDataManager = ViewModelProvider(requireActivity()).get(MoviesDataManager::class.java)
         bindMoviesRecyclerView()
-        val sharedPref = SignInRepository.sharedPrefs()
-        val userName =
-            sharedPref.getString(SignInFragment.USER_NAME_KEY, "")
-        val userImage = sharedPref.getString(SignInFragment.USER_IMAGE_KEY,"")
-        binding?.contentLayout?.toolbarImage?.let {
-            Glide.with(this).load(userImage).placeholder(R.drawable.ic_account_circle_white_24dp)
```
Comment (jenndotcodes) [inline Week4/app/src/main/java/com/kulloveth/moviesapp/ui/movies/MoviesFragment.kt:69]: Nice work using Glide!

File: Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt [hunk 5]
```diff
@@ -46,9 +47,8 @@ abstract class MovieDatabse : RoomDatabase() {
...
-                    Injection.provideRepository.insertAllMovie(
```
Comment (jenndotcodes) [inline Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt:50]: Nice work on using Injection.

File: Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt [hunk 6]
```diff
@@ -46,9 +47,8 @@ abstract class MovieDatabse : RoomDatabase() {
...
-                        MoviesDataManager.movieList)
+                CoroutineScope(context = Dispatchers.IO).launch {
+                    Injection.provideRepository.insertAllMovie(MoviesDataManager.movieList)
                 }
```
Comment (jenndotcodes) [inline Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt:52]: Nice job removing allow main thread queries.

**Review by jenndotcodes — COMMENTED** [review]
Really nice work on this project, overall you've done a great job, it works really well and you've provided suspend functions and utilized coroutines and many architechture concepts. I noticed you used Flow. There is one place that GlobalScope is used and that is generally to be avoided. The code is commented and readable and of good style. I am awarding 'good job', keep up the great work!

File: Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt (hunk 2 above)
Comment (kulloveth) [reply Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt:49]: Oh this skipped my mind ofCourse there are lot of ways to do that, Thanks

Comment (kulloveth) [reply Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt:49]: ` private fun builDatabase() = Room.databaseBuilder(
            context,
            MovieDatabse::class.java,
            "movie_database"
        ).addCallback(object : RoomDatabase.Callback() {
            override fun onCreate(db: SupportSQLiteDatabase) {
                super.onCreate(db)
                CoroutineScope(context = Dispatchers.IO).launch {
                    Injection.provideRepository.insertAllMovie(MoviesDataManager.movieList)
                }
            }
        }).build()
    }`
This is what I have in my local,I guess I forgot to update that

File: Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt (hunk 6 above)
Comment (kulloveth) [reply Week4/app/src/main/java/com/kulloveth/moviesapp/db/room/MovieDatabse.kt:52]: OH looking at this I actually removed the global scope before I submitted, kindly take a look, I guess the GlobalScope  was an initial commit

File: Week4/app/src/main/java/com/kulloveth/moviesapp/ui/movies/MoviesFragment.kt (hunk 4 above)
Comment (kulloveth) [reply Week4/app/src/main/java/com/kulloveth/moviesapp/ui/movies/MoviesFragment.kt:69]: I had issues with Glide though so my final commits are using Picasso instead, even the Glide dependency is not there anymore

**Pull request merged**